snappass/templates/deployment.yaml

```diff
@@ -0,0 +1,128 @@
+          - name: REDIS_PORT
+            value: "{{ .Values.redis.redisPort }}"
+          {{- else }}
+          - name: REDIS_HOST
+            value: "{{ $redis.host }}"
+          - name: REDIS_PORT
+            value: "{{ $redis.port }}"
+          {{- end }}
+          command: [ "/bin/sh", "-c", "until nc -zv ${REDIS_HOST} ${REDIS_PORT} -w1; do sleep 1; done" ]
+        {{- end }}
+      containers:
+        - name: {{ .Chart.Name }}
+          securityContext:
+            {{- toYaml .Values.securityContext | nindent 12 }}
+          image: "{{ .Values.image.repository }}:{{ .Values.image.tag }}"
+          imagePullPolicy: {{ .Values.image.pullPolicy }}
+          env:
+          {{- if $snappass.debug.enabled }}
+          - name: DEBUG
+            value: "true"
+          {{- end }}
+          {{- if not $snappass.ssl.enabled }}
+          - name: NO_SSL
+            value: "true"
+          {{- end }}
+          {{- if $snappass.urlPrefix }}
+          - name: URL_PREFIX
+            value: "{{ $snappass.urlPrefix }}"
+          {{- end }}
+          {{- if $snappass.staticUrl }}
+          - name: STATIC_URL
+            value: "{{ $snappass.staticUrl }}"
+          {{- end }}
+          - name: SECRET_KEY
+            valueFrom:
+              secretKeyRef:
+                name: {{ if $snappass.existingSecret }}{{ $snappass.existingSecret }}{{ else }}{{ include "snappass.fullname" . }}{{ end }}
+                key: secretKey
+          {{- if .Values.redis.enabled }}
+          - name: REDIS_HOST
+            value: "{{ template "snappass.redis.fullname" . }}-master"
+          - name: REDIS_PORT
+            value: "{{ .Values.redis.redisPort }}"
+          - name: REDIS_PASSWORD
+            valueFrom:
+              secretKeyRef:
+                name: "{{ if .Values.redis.existingSecret }}{{ .Values.redis.existingSecret }}{{ else }}{{ template "snappass.redis.fullname" . }}{{ end }}"
+                key: "{{ if and .Values.redis.existingSecret .Values.redis.existingSecretPasswordKey  }}{{ .Values.redis.existingSecretPasswordKey }}{{ else }}redis-password{{ end }}"
+          {{- else }}
+          - name: REDIS_HOST
+            value: "{{ $redis.host }}"
+          - name: REDIS_PORT
+            value: "{{ $redis.port }}"
+          - name: REDIS_PASSWORD
+            valueFrom:
+              secretKeyRef:
+                name: "{{ if $redis.existingSecret }}{{ $redis.existingSecret }}{{ else }}{{ include "snappass.fullname" . }}-redis{{ end }}"
+                key: "redis-password"
+          {{- end }}
+          - name: SNAPPASS_REDIS_DB
+            value: "{{ $redis.db }}"
+          {{- if $redis.prefix }}
+          - name: REDIS_PREFIX
+            value: "{{ $redis.prefix }}"
+          {{- end }}
+          command: ["/bin/sh"]
+          args: ["-c", "REDIS_URL=redis://redis:$(REDIS_PASSWORD)@$(REDIS_HOST):$(REDIS_PORT)/$(SNAPPASS_REDIS_DB) snappass"]
```

Comment:
Could please test if signals (i.e. SIGTERM) are properly forwarded to the process?
If not we might use `exec snappass`

Reply:
SnapPass itself can't handle signals properly. Anyways, I added `exec env`.